tensorbay/opendataset/HalpeFullBody/loader.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name
+
+"""Dataloader of the Halpe Full-Body Human Keypoints and HOI-Det dataset."""
+
+import json
+import os
+from typing import Any, Dict
+
+from ...dataset import Data, Dataset
+from ...label import Classification, LabeledBox2D, LabeledKeypoints2D
+
+DATASET_NAME = "Halpe Full-Body Human Keypoints and HOI-Det"
+
+_SEGMENT_SPLIT = (
+    ("train", "halpe_train_v1.json", os.path.join("hico_20160224_det", "images", "train2015")),
+    ("val", "halpe_val_v1.json", "val2017"),
+)
+_KEYPOINTS_NUM = 408
+
+
+def HalpeFullBody(path: str) -> Dataset:
+    """Dataloader of the Halpe Full-Body Human Keypoints and HOI-Det dataset.
+
+    Arguments:
+        path: The root directory of the dataset.
+            The folder structure should be like::
+
+                <path>
+                    halpe_train_v1.json
+                    halpe_val_v1.json
+                    hico_20160224_det/
+                        images/
+                            train2015/
+                                HICO_train2015_00000001.jpg
+                                ...
+                    val2017/
+                        000000000139.jpg
+                        ...
+
+    Returns:
+        Loaded `Dataset` object.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+
+    for mode, label_file, image_directory in _SEGMENT_SPLIT:
+        segment = dataset.create_segment(mode)
+
+        with open(os.path.join(root_path, label_file), "r", encoding="utf-8") as fp:
+            annotations = json.load(fp)
+
+        for annotation, image in zip(annotations["annotations"], annotations["images"]):
+
+            if mode == "train":
+                image_path = os.path.join(root_path, image_directory, image["file_name"])
+            else:
+                image_path = os.path.join(
+                    root_path, image_directory, f"{annotation['image_id']}.jpg".zfill(16)
+                )
+
+            segment.append(_get_data(image_path, annotation, image))
+
+    return dataset
+
+
+def _get_data(image_path: str, annotation: Dict[str, Any], image: Dict[str, Any]) -> Data:
+    data = Data(image_path)
+
+    raw_keypoints = annotation["keypoints"]
+    keypoints = []
```

Comment:
init a `LabeledKeypoints` then append keypoints to it

Reply:
reference: 
https://github.com/linjiX/tensorbay-python-sdk/blob/525b1007708cb67b04391c930b6afa04f8276c48/tensorbay/opendataset/LeedsSportsPose/loader.py#L54-L58